src/main/java/io/confluent/connect/elasticsearch/bulk/BulkProcessor.java

```diff
@@ -122,20 +125,35 @@ public Thread newThread(Runnable r) {
     };
   }
 
-  private Runnable farmerTask() {
-    return new Runnable() {
-      @Override
-      public void run() {
-        log.debug("Starting farmer task");
-        try {
-          while (!stopRequested) {
-            submitBatchWhenReady();
+  // visible for testing
+  Runnable farmerTask() {
+    return () -> {
+      log.debug("Starting farmer task");
+      try {
+        List<Future<BulkResponse>> futures = new ArrayList<>();
+        while (!stopRequested) {
+          // submitBatchWhenReady waits for lingerMs so we won't spin here unnecessarily
+          futures.add(submitBatchWhenReady());
+
+          // after we submit, look at any previous futures that were completed and call get() on
+          // them so that exceptions are propagated.
+          List<Future<BulkResponse>> unfinishedFutures = new ArrayList<>();
+          for (Future<BulkResponse> f : futures) {
+            if (f.isDone()) {
+              log.debug("Calling get on completed future");
```

Comment:
Since `get()` is guaranteed to return quickly, maybe we can add print some info from the response object here?

```suggestion
              log.debug("Bulk request completed with status {}", f.get());
```

`BulkResponse` would need a `toString()` method.